Allow SCRIP reader to respect units w/r/t radians

zarzycki opened the pull request:
## Overview
- Currently, the SCRIP reader does not respect units attached to relevant lat/lon vars. That is, it assumes degrees. For 90% of grids, this is probably the case. MPAS, by default, writes SCRIP files internally in radians, which are easily convertible using NCO or other external scripts, but it breaks uxarray currently (well, uxarray runs fine since |radians| < |degrees| but then projects fields in a very small location near 0, 0).
- This just adds a simple check to see if "radians" (or "radians-adjacent") units are attached to the grid_XX_lat/lon vars on the SCRIP nc file and, if so, converts upon loading.

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [x] Adequate tests are created if there is new functionality
- [ ] Tests cover all possible logical paths in your function
- [x] Tests are not too basic (such as simply calling a function and nothing else)

**Documentation**
- [x] Docstrings have been added to all new functions
- [N/A] Docstrings have updated with any function changes
- [ ] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [N/A] User functions have been added to `docs/user_api/index.rst`

zarzycki committed:
Allow SCRIP reader to respect units wrt to radians

Sevans711 commented:
Hi @zarzycki, thank you for drafting this PR! It has been a while, so I wanted to check, can you confirm whether this still needs to be fixed, in the latest version of uxarray?

If yes, it would be extremely helpful if you might be able to provide a tiny example (~1 MB or less would be ideal) where the issue occurs.

Then, the remaining steps would be:
1. make an issue report (https://github.com/UXARRAY/uxarray/issues) for this issue.
2. add the tiny example file, somewhere in uxarray/test/meshfiles/scrip.
3. make a test (in uxarray/test/io/test_scrip.py) to check units are handled properly after your fix.
4. merge main into this branch (via the "Update branch" button) and make sure all tests pass.
5. someone else reviews the changes and merges the fix into main.

To make things as easy as possible, if you can confirm this is still an issue and can provide a tiny example file here, I'd be happy to take over any/all of steps 1 through 4 if you want, and then hopefully we could get this fix included in the next uxarray release!

zarzycki committed:
Add test file and unit test for SCRIP radian coordinate handling

zarzycki commented:
Thanks @Sevans711. I can confirm the bug is still present in uxarray 2026.6.0.

A SCRIP file with coordinates stored in radians is read without conversion; e.g. face_lon returns [0.17, 0.35] instead of [10.0, 20.0]. 
  
I've opened issue #1564 and pushed a tiny (synthetic) SCRIP file (test/meshfiles/scrip/scrip_radians.nc) along with a test (test_scrip_radians_units in test/io/test_scrip.py) that reproduces the failure on unpatched uxarray and passes with this fix. I remain a Github novice-ish when issuing PRs for a project I am not running point on, so please let me know if other things are needed from me (Claude helped here :)

zarzycki committed:
Merge branch 'main' into fix-scrip-radians-units

Sevans711 committed:
minor formatting changes for scrip radians fix

Moves meshfiles/scrip/scrip_radians.nc to meshfiles/scrip/scrip_radians/scrip_radians_grid.nc to match style of other meshfiles naming schemes.

Renames the new _scrip._convert_to_degrees() to _scrip._values_in_degrees(). It doesn't always convert; and it also does more than just converting, because it gives numpy array from DataArray. Clarified docstring.

ran the following, so it should now pass ruff checks: pre-commit run --all-files

Sevans711 committed:
Merge branch 'main' into fix-scrip-radians-units

Sevans711 commented:
The small test case looks good, I was also able to confirm it fails on `main` but succeeds after your fix. Thank you @zarzycki for all your efforts here! I made some minor formatting/style changes but this looks good to me now and I think it is ready to merge to main.

Pinging @rajeeja or @erogluorhan to review if you get a chance! Additionally, I notice now that the PR default form (see original message) includes the task "Internal functions have a preceding underscore (_) and have been added to docs/internal_api/index.rst." Can you confirm whether this is necessary? I don't actually see the `docs/internal_api/index.rst` file, maybe it has been renamed or removed?

rajeeja reviewed (approved):
Verified locally — radians convert correctly, degrees files unaffected, SCRIP tests pass. Thanks @zarzycki.

rajeeja committed:
Merge branch 'main' into fix-scrip-radians-units
